### ISSUE TITLE
Support for returning LRSchedulerConfig on LightningModule.configure_optimizers

### ISSUE THREAD

**Issue by sarpaykent** [Lightning-AI/lightning#13569]
## 🚀 Feature

Currently, LRSchedulerConfig type is not supported by [_configure_schedulers_automatic_opt](https://github.com/Lightning-AI/lightning/blob/61c28cb428a13c2aea6d7f3f55e0f00431a4ea4e/src/pytorch_lightning/core/optimizer.py#L263). 

### Motivation

I created a custom `ReduceLROnPlateau` class that adds warmup epochs before `ReduceLROnPlateau`. However, the `ReduceLROnPlateau` scheduler is only supported for the `torch.optim.lr_scheduler.ReduceLROnPlateau` class. This is [implemented](https://github.com/Lightning-AI/lightning/blob/61c28cb428a13c2aea6d7f3f55e0f00431a4ea4e/src/pytorch_lightning/core/optimizer.py#L288) by setting `reduce_on_plateau` property based on `isinstance` check based on the type of the scheduler. The check causes a problem using a custom scheduling class with similar properties as 'ReduceLROnPlateau'. There is a need for an alternative mechanism to notify the `_configure_schedulers_automatic_opt` that the scheduler is a derivative of ReduceLROnPlateau (not necessarily a subclass).

### Pitch

If desired, users can gain more control by bypassing this mechanism by directly returning `LRSchedulerConfig` from `configure_optimizers` rather than a dictionary. This can be implemented by simple two-line addition to [here](https://github.com/Lightning-AI/lightning/blob/61c28cb428a13c2aea6d7f3f55e0f00431a4ea4e/src/pytorch_lightning/core/optimizer.py#L311):

```python
...
elif isinstance(scheduler, LRSchedulerConfig):
    config = scheduler
...
```

This is a very straightforward solution to this problem. I'm not aware of any side effects. It is fair to assume it is an advanced use case and does not break any compatibility with the existing code base. 

cc @borda @rohitgr7 @akihironitta

**Comment by carmocca** [comment]
We try to avoid that the users have to import and use these internal structures. However, we could make it so we check if the dictionary already has `reduce_lr_on_plateau` set and use the existing value if so.

Would you like to work on a PR with that adds support for this?

**Comment by MrShevan** [comment]
Hi, could you assign it to me?)
